### PR TITLE
Add connection switching to gcp_backend

### PR DESCRIPTION
**Pull request by slevenick**
Allow for magic modules generated resources by adding gcp api http connection management

### CODE REVIEW

File: libraries/gcp_backend.rb [hunk 1]
```diff
@@ -178,3 +186,203 @@ def camel_case(data)
...
+    def authorize_http(req)
+      req.extend TokenProperty
+      auth = {}
+      @authorization.apply!(auth)
+      req['Authorization'] = auth[:authorization]
+      req.token = auth[:authorization].split(' ')[1]
+      req
+    end
+  end
+      # Extension methods to enable retrieving the authentication token.
```
Comment (rambleraptor) [inline libraries/gcp_backend.rb]: Can you indent this in-line with line 384?

File: libraries/gcp_backend.rb [hunk 2]
```diff
@@ -178,3 +186,203 @@ def camel_case(data)
...
+    end
+
+    private
+
+    def generate_user_agent
+      "inspec-google/1.0.0"
+    end
+  end
+  class Authorization
+  # A class to aquire credentials and authorize Google API calls.
```
Comment (rambleraptor) [inline libraries/gcp_backend.rb]: Can you put this above line 330? (That seems to be the canonical way of putting a comment describing a class)

File: libraries/gcp_backend.rb [hunk 3]
```diff
@@ -178,3 +186,203 @@ def camel_case(data)
...
+        raise "Missing variable :#{v} in #{var_data} on #{caller.join("\n")}}"
+      end
+      template.gsub!(/{{#{v}}}/, CGI.escape(var_data[v].to_s))
+    end
+    template
+  end
+end
+
+module Network
+    # A handler for authenticated network request
```
Comment (rambleraptor) [inline libraries/gcp_backend.rb]: Put above `module Network`

File: libraries/gcp_backend.rb [hunk 4]
```diff
@@ -17,6 +19,12 @@ def initialize(opts)
     @opts = opts
     # ensure we have a GCP connection, resources can choose which of the clients to instantiate
     @gcp = inspec.backend
+
+    # Magic Modules generated resources use an alternate transport method
+    # In the future this will be moved into the train-gcp plugin itself
+    if opts[:use_http_transport]
```
Comment (rambleraptor) [inline libraries/gcp_backend.rb]: How does this option get passed in? What happens if it doesn't get passed in?

Comment (slevenick) [reply libraries/gcp_backend.rb]: The option is passed in by MM generated resources that inherit this class. They call add that flag to their opts when they pass it to super().

If it isn't passed in we assume the resource does not use the REST APIs directly, and instead will use the SDKs, which is the current functionality.

Comment (rambleraptor) [reply libraries/gcp_backend.rb]: Ah, gotcha. That's a good plan.

**Pull request merged**